### DISCUSSION TITLE
Code coverage problem for <script setup>

### DISCUSSION THREAD

**Discussion by Xeyos88** [vitest-dev/vitest#5120] [General]
How is it possible that the <script setup> results as an uncovered branch?

![Untitled](https://github.com/vitest-dev/vitest/assets/16436299/86aeaf56-0b35-4ded-af43-41fccbeed7e7)

**Comment by BIGScream** [comment]
I have the same issue with some first lines - this file was 100% covered weeks ago and has no changes since then.
![Code_coverage_report_for_wkfs-nuxt_components_layout_HeaderSubMenu_vue](https://github.com/vitest-dev/vitest/assets/70846921/5b5e0d20-0443-44de-8c24-a4647536e304)
I don't find any solutions for this

**Reply by BIGScream** [reply]
problem is not solved by updating vitest, @vitest/coverage-c8, @vue/test-utils, @vitejs/plugin-vue, @nuxt/test-utils-edge to latest versions. I think its a dependency from one of those...

**Reply by unherz** [reply]
I had a very similar problem with the same output and hint.
The problem in my case was that an conditional rendering was not covered.

Example:
```
        <span class="c-card__date" v-if="item.date">
          {{ $t('card.date') }}
          <time :datetime="item.date.toLocaleDateString('en-US')">{{ item.date.toLocaleDateString('en-US') }}</time>
        </span>
```
I had no test case specified for a missing date on my item.

Therefore the reporting in this case should be more interpreted as "these is something not covered in your template".
But the report itself is very bad for this case.

**Reply by bartduisters** [reply]
To add to the other comments.
@unherz is right that it should be interpreted as 'there is something not covered in the template'.

But, I'd expect the coverage tooling to just highlight the line in the template. Like this:
![Screenshot 2024-08-12 at 14 30 13](https://github.com/user-attachments/assets/c77241d1-398d-44ef-9ecc-0a8de9f9b91d)

In my specific use case, I did not have the use case covered where `show` is false on the root element in the template.


I've also noticed that when adding an empty line as first line, it does not complain at all, it will claim 100% coverage.

![Screenshot 2024-08-12 at 14 48 29](https://github.com/user-attachments/assets/deb1dc76-fed4-44ff-93ba-ad97d6695dc7)
![Screenshot 2024-08-12 at 14 48 18](https://github.com/user-attachments/assets/5b4d0628-4cb7-435f-9e51-6311b5b18a76)
![Screenshot 2024-08-12 at 14 47 59](https://github.com/user-attachments/assets/2dc6f59b-c80f-445e-96f9-18635e3cb2f5)
![Screenshot 2024-08-12 at 14 47 43](https://github.com/user-attachments/assets/6102962a-5930-4ce2-ad0f-a2d09b937d62)
![Screenshot 2024-08-12 at 14 47 20](https://github.com/user-attachments/assets/12ed6e38-80c0-4b3a-9849-c4ae0b2e4788)
![Screenshot 2024-08-12 at 14 46 59](https://github.com/user-attachments/assets/a59d38cb-4c2b-4b1e-91a4-21c3446bcdc0)

**Reply by AriPerkkio** [reply]
This was fixed in 
- https://github.com/vuejs/core/issues/13261

**Reply by AntonioVentilii** [reply]
@AriPerkkio how was that exactly solved? I have the latest versions of everything and still the same issue. May you please help us decipher the solution?

**Reply by AriPerkkio** [reply]
@AntonioVentilii in the linked issue it's described how Vue compiler incorrectly added generated code in the source maps that were pointing to the first line of the source file. This generated line wasn't always executed when running the code, so coverage tools would mark it as uncovered. It was fixed by https://github.com/vuejs/core/pull/13285 and released in `@vue/compiler-core@3.5.15`: https://github.com/vuejs/core/blob/main/CHANGELOG.md#3515-2025-05-26

**Comment by AntonioVentilii** [comment]
Any solution for this please?

**Comment by BIGScream** [comment]
i found out, its often v-if's or events that are not tested. Why they are not marked in their lines, i can't catch

**Comment by AntonioVentilii** [comment]
@BIGScream not sure that is just that. My component is ONLY this:

```svelte
<script lang="ts">
	import TermsOfUse from '$lib/components/terms-of-use/TermsOfUse.svelte';
</script>

<TermsOfUse />
```


and i have

<img width="1476" height="338" alt="Screenshot 2025-09-18 at 13 09 44" src="https://github.com/user-attachments/assets/aede0cab-6a61-4751-a3c7-d6f654fb73e2" />
